### PR TITLE
Use Context CL whenever it is applicable

### PR DESCRIPTION
**Pull request by artembilan**
* Some code style refactoring: mostly lambdas to method references

### CODE REVIEW

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java [hunk 1]
```diff
@@ -617,7 +618,7 @@ private ChannelProxy getCachedChannelProxy(ChannelCachingConnectionProxy connect
 		else {
 			interfaces = new Class<?>[] { ChannelProxy.class };
 		}
-		return (ChannelProxy) Proxy.newProxyInstance(ChannelProxy.class.getClassLoader(),
+		return (ChannelProxy) Proxy.newProxyInstance(ClassUtils.getDefaultClassLoader(),
 				interfaces, new CachedChannelInvocationHandler(connection, targetChannel, channelList,
```
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java]: According to Andy's comments, we don't really need it here, right?

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java [hunk 2]
```diff
@@ -1131,7 +1132,7 @@ protected void initializeProxy(Object delegate) {
 		}
 		factory.addInterface(ContainerDelegate.class);
 		factory.setTarget(delegate);
-		this.proxy = (ContainerDelegate) factory.getProxy(ContainerDelegate.class.getClassLoader());
+		this.proxy = (ContainerDelegate) factory.getProxy(ClassUtils.getDefaultClassLoader());
```
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java]: ditto

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java [hunk 3]
```diff
@@ -583,8 +584,9 @@ private void checkMissingQueues(String[] queueNames) {
 				 * Use reflection to avoid class tangles.
 				 */
 				try {
-					Class<?> clazz = ClassUtils.forName("org.springframework.amqp.rabbit.core.RabbitAdmin",
-							getClass().getClassLoader());
+					Class<?> clazz =
+							ClassUtils.forName("org.springframework.amqp.rabbit.core.RabbitAdmin",
+									ClassUtils.getDefaultClassLoader());
```
Comment (garyrussell) [inline spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java]: and here

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java (hunk 1 above)
Comment (artembilan) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java]: I'd rather have these classes loaded in the `RestartClassLoader` to be able to pick the changes up, than once sticked with the `Launcher$AppClassLoader` instead. However I might be wrong and will change this back, if you insist and have some simple explanation why we don't need this change here and there.

My understanding it is OK for `static` blocks, but if we do something at runtime would be better to rely on the context classloader. 

We can use `getApplicationContext().getClassLoader()` instead though...

File: spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java (hunk 2 above)
Comment (artembilan) [reply spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java]: The same here about CL from AC.

**Pull request merged**